tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailbox.scala

```diff
@@ -43,10 +44,56 @@ object TeamMailboxName {
       .map(TeamMailboxName(_))
 }
 
+object TeamMailboxFQDN {
+  private val MAXIMUM_CHARACTERS_OF_MAIL_ADDRESS = 320
+  type TeamMailboxFQDNType = String Refined TeamMailboxFQDNConstraint
+  private val charMatcher: CharMatcher = CharMatcher.inRange('a', 'z')
+    .or(CharMatcher.inRange('0', '9'))
+    .or(CharMatcher.inRange('A', 'Z'))
+    .or(CharMatcher.is('_'))
+    .or(CharMatcher.is('-'))
+    .or(CharMatcher.is('#'))
+    .or(CharMatcher.is('.'))
+    .or(CharMatcher.is('@'))
+
+  case class TeamMailboxFQDNConstraint()
+
+  implicit val validateTeamMailboxName: Validate.Plain[String, TeamMailboxFQDNConstraint] =
+    Validate.fromPredicate(s => s.nonEmpty && s.length <= MAXIMUM_CHARACTERS_OF_MAIL_ADDRESS && charMatcher.matchesAllOf(s),
+      s => s"'$s' contains some invalid characters. Should be [#a-zA-Z0-9-_.@] and no longer than 320 chars",
+      TeamMailboxFQDNConstraint())
+
+  def validate(string: String): Either[IllegalArgumentException, TeamMailboxFQDNType] =
+    refined.refineV[TeamMailboxFQDNConstraint](string)
+      .left
+      .map(new IllegalArgumentException(_))
+
+  def toTeamMailbox(value: TeamMailboxFQDNType): Either[IllegalArgumentException, TeamMailbox] =
+    validate(value)
+      .map(TeamMailboxFQDN(_))
+      .map(_.toTeamMailbox())
+}
+
 case class TeamMailboxName(value: TeamMailboxNameType) {
   def asString(): String = value.value
 }
 
+case class TeamMailboxFQDN(value: TeamMailboxFQDNType) {
+  def asString(): String = value.value
+
+  def toTeamMailbox(): TeamMailbox = {
+    val atPosition = value.indexOf('@')
+    if (atPosition < 0) {
+      throw new IllegalArgumentException("Missing '@' in mailbox FQDN")
+    }
+    val localPart = value.substring(0, atPosition)
+    val domainPart = value.substring(atPosition + 1)
+    TeamMailbox(domain = Domain.of(domainPart),
+      mailboxName = TeamMailboxName.fromString(localPart)
+        .fold((e: IllegalArgumentException) => throw e, (x: TeamMailboxName) => x))
+  }
+}
+
```

Comment:
IMO TeamMailboxFQDN is not needed and we should work only with TeamMailbox abstraction...